### PR TITLE
24 - Iterative search

### PR DESCRIPTION
**Pull request by npauzenga**
It took a little time but I was able to implement binary search from memory in this checkpoint. Could definitely use more practice implementing search and sort algorithms. Actually, this weekend I'll take a stab at a recursive version of binary search!

### CODE REVIEW

**Comment by npauzenga** [comment]
@Enriikke Ha I _really seriously_ need to configure Hound and Rubocop. I've updated the quotes to double in this branch so once it's merged, those warning should cease. Rubocop was complaining about the parameter alignment just like Hound is above but I haven't been able to make anyone happy (might be line breaks in there somewhere that I'm missing maybe).

**Comment by npauzenga** [comment]
Refactored with recursive search. Probably a cleaner way to do this without changing the parameters but hey, it seems to work :grin:

File: models/address_book.rb [hunk 1]
```diff
@@ -31,25 +33,25 @@ def import_from_csv(file_name)
...
-    upper = entries.length - 1
-
-    while lower <= upper
-      mid = (lower + upper) / 2
-      mid_name = entries[mid].name
+  def binary_search(name, lower, upper)
+    mid = (upper + lower) / 2
+    
+    if lower >= upper || entries[mid].name == name
+      entries[mid].name == name ? entries[mid] : nil
```
Comment (enriikke) [inline models/address_book.rb]: It looks like you are checking the same condition twice. Maybe you can refactor it to have use a "guard clause" instead :smiley: :

``` ruby
mid = (upper + lower) / 2
return entries [mid] if entries[mid].name == name

# Rest of the code
...
```

**Comment by enriikke** [comment]
Hey @npauzenga! Everything looks great! Awesome job taking at stab at recursion :smiley:. I made a small comment on the recursive binary search method but you can merge without it :+1: :shipit:

**Pull request merged**